### PR TITLE
Node will get all neighbors state instead of just voting neighbors

### PR DESCRIPTION
**Pull request by yilunzhang**
### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: consensus/state.go [hunk 1]
```diff
@@ -100,14 +100,21 @@ func (consensus *Consensus) getNeighborConsensusState(neighbor *node.RemoteNode)
...
+func (consensus *Consensus) getVotingNeighborsConsensusState() (map[string]*pb.GetConsensusStateReply, error) {
+	allNeighbors := consensus.localNode.GetNeighbors(nil)
+	votingNeighbors := consensus.localNode.GetVotingNeighbors(nil)
+	if len(votingNeighbors) < minConsensusStateNeighbors {
+		votingNeighbors = allNeighbors
 	}
-	allStates := make(map[string]*pb.GetConsensusStateReply, len(allNeighbors))
+	votingNeighborsMap := make(map[string]struct{}, len(votingNeighbors))
+	for _, neighbor := range votingNeighbors {
+		votingNeighborsMap[neighbor.GetID()] = struct{}{}
```
Comment (bufrr) [inline consensus/state.go:114]: Why we need to initialize `votingNeighborsMap` , we may have key with empty value, which previous code wont have

Comment (yilunzhang) [reply consensus/state.go:114]: This is map not slice, make will not create empty value

File: consensus/state.go [hunk 2]
```diff
@@ -100,14 +100,21 @@ func (consensus *Consensus) getNeighborConsensusState(neighbor *node.RemoteNode)
...
+	}
+
+	states := make(map[string]*pb.GetConsensusStateReply, len(allNeighbors))
```
Comment (bufrr) [inline consensus/state.go:117]: maybe same length as `votingNeighbors`

Comment (yilunzhang) [reply consensus/state.go]: Fixed

**Review by bufrr — APPROVED** [review]

**Pull request merged**